src/client/Microsoft.Identity.Client/Utils/JsonHelper.cs

```diff
@@ -31,14 +32,19 @@ internal static string SerializeToJson<T>(T toEncode)
 #endif
         }
 
+#if SUPPORTS_SYSTEM_TEXT_JSON
+        [RequiresUnreferencedCode("Calls System.Text.Json.JsonSerializer.Deserialize(String, Type, JsonSerializerOptions)")]
```

Comment:
This is needed when using JsonSerializer.Deserialize(json, typeof(T), MsalJsonSerializerOptions.Options to avoid broken functionality when using trimming.

Message:
Warning	IL2026	Using member 'System.Text.Json.JsonSerializer.Deserialize(String, Type, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. JSON serialization and deserialization might require types that cannot be statically analyzed. Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make sure all of the required types are preserved.

Reply:
Im going to keep trying different things to see if I can find a better solution but at least this one works.

Reply:
Seems like Context class is better to use... They do use both Options and Context in the samples. There are different modes of source generation, so maybe we need to set some specific properties for Options to work the same as Context.
https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation

Reply:
Trimming is important, we have customers using it in public client world. 

If we can't find a good solution at JSON level, consider having separate logic for AAD response / B2C response.

Reply:
I agree with Bogdan that if the response format is different for AAD and B2C, it would be good to have separate deserialization logic for each.

Reply:
Removed Il trimming attribute and using a better solution.